cli/azd/cmd/env.go

```diff
@@ -150,6 +168,293 @@ func (e *envSetAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 	return nil, nil
 }
 
+func newEnvSetSecretFlags(cmd *cobra.Command, global *internal.GlobalCommandOptions) *envSetSecretFlags {
+	flags := &envSetSecretFlags{}
+	flags.Bind(cmd.Flags(), global)
+
+	return flags
+}
+
+type envSetSecretFlags struct {
+	internal.EnvFlag
+	global *internal.GlobalCommandOptions
+}
+
+func (f *envSetSecretFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
+	f.EnvFlag.Bind(local, global)
+	f.global = global
+}
+
+type envSetSecretAction struct {
+	console            input.Console
+	azdCtx             *azdcontext.AzdContext
+	env                *environment.Environment
+	envManager         environment.Manager
+	flags              *envSetFlags
+	args               []string
+	prompter           prompt.Prompter
+	kvService          keyvault.KeyVaultService
+	entraIdService     entraid.EntraIdService
+	subResolver        account.SubscriptionTenantResolver
+	userProfileService *azapi.UserProfileService
+}
+
+func (e *envSetSecretAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+
+	if len(e.args) < 1 {
+		return nil, fmt.Errorf(
+			"no <name> provided. Please provide a name as argument like: 'azd env set <name>'")
+	}
+	secretName := e.args[0]
+
+	createNewStrategy := "Create a new Key Vault secret"
+	selectExistingStrategy := "Select an existing Key Vault secret"
+	setSecretStrategies := []string{createNewStrategy, selectExistingStrategy}
+	selectedStrategyIndex, err := e.console.Select(
+		ctx,
+		input.ConsoleOptions{
+			Message:      "Select how you want to set " + secretName,
+			Options:      setSecretStrategies,
+			DefaultValue: createNewStrategy,
+			Help: "When creating a new Key Vault secret, you can either create a new Key Vault account or" +
+				" pick an existing one. A Key Vault secret belongs to a Key Vault account.",
```

Comment:
Key Vault:
https://styleguides.azurewebsites.net/Styleguide/Read?id=2696&topicid=72112

```suggestion
			Help: "When creating a new Key Vault secret, you can either create a new Key Vault or" +
				" pick an existing one. A Key Vault secret belongs to a Key Vault.",
```

Reply:
same elsewhere in the text